File: kora-app-annotation-processor/src/test/java/ru/tinkoff/kora/kora/app/annotation/processor/app/AppWithFactories10.java
package ru.tinkoff.kora.kora.app.annotation.processor.app;

import ru.tinkoff.kora.common.KoraApp;
import ru.tinkoff.kora.common.annotation.Root;

import java.io.Closeable;

@KoraApp
public interface AppWithFactories10 {
    @Root
    default Object mock1(Closeable object) {
        return new Object();
    }
}

File: kora-app-symbol-processor/src/test/kotlin/ru/tinkoff/kora/kora/app/ksp/app/AppWithFactories10.kt
package ru.tinkoff.kora.kora.app.ksp.app

import ru.tinkoff.kora.common.KoraApp
import ru.tinkoff.kora.common.annotation.Root
import java.io.Closeable

@KoraApp
interface AppWithFactories10 {
    @Root
    fun mock1(registry: Closeable) = Any()
}
